lib/CFL/CFGNormalizer.cpp

```diff
@@ -237,7 +234,7 @@ GrammarBase::Production CFGNormalizer::getFilledProd(GrammarBase::Production &pr
 ///and expand to productions set
 ///e.g Xi -> Y Zi with Xi i = 0, 1, Yi i = 0,2
 ///Will get {X0 -> Y Z0, X1 -> Y Z1, X2 -> Y Z2}
-GrammarBase::Productions CFGNormalizer::getFilledProductions(GrammarBase::Production &prod, const Map<CFLGrammar::Kind,  Set<CFLGrammar::Attribute>>& kind2AttriMap, CFLGrammar *grammar)
+GrammarBase::Productions CFGNormalizer::getFilledProductions(GrammarBase::Production &prod, const NodeSet& nodeSet, CFLGrammar *grammar)
```

Comment:
Avoid copy construction for Production and Productions. Also need to carefully check other places for parameters and returns.